src/components/encoding-pane/field-customizer.tsx

```diff
@@ -12,35 +14,136 @@ export interface FieldCustomizerProps extends ActionHandler<SpecEncodingAction>
+    }
+  ],
+  'Stack': [
+    {
+      prop: 'stack',
+      nestedProp: 'stack'
+    }
+  ]
+};
 
+const POSITIONAL_ENCODING_NON_QUANTITATIVE = {
+  'Scale': [
+    {
+      prop: 'scale',
+      nestedProp: 'type'
+    }
+  ],
+  'Axis': [
+    {
+      prop: 'axis',
+      nestedProp: 'orient'
+    },
+    {
+      prop: 'axis',
+      nestedProp: 'title'
+    }
+  ]
+};
+
+// TODO: Mark channel encoding
+const CHANNEL_ENCODING_MAP = {
+  'position': {
+    "quantitative": POSITIONAL_ENCODING_QUANTITATIVE,
+    "ordinal": POSITIONAL_ENCODING_NON_QUANTITATIVE,
+    "temporal": POSITIONAL_ENCODING_NON_QUANTITATIVE,
+    "nominal": POSITIONAL_ENCODING_NON_QUANTITATIVE
+  },
+  'mark': {
+  },
+};
+
+export class FieldCustomizerBase extends React.PureComponent<FieldCustomizerProps, {}> {
   public render() {
     const {shelfId, handleAction, fieldDef} = this.props;
+    const keys = Object.keys(this.getEncodingTypeForChannel());
     return (
-      <div styleName='customizer-container'>
-        {this.customizableProps().map(customizableProp => {
-          const {prop, nestedProp} = customizableProp;
-          return (
-            <PropertyEditor
-              key={JSON.stringify({prop, nestedProp})}
-              prop={prop}
-              nestedProp={nestedProp}
-              shelfId={shelfId}
-              fieldDef={fieldDef}
-              handleAction={handleAction}
-            />
-          );
-        })}
+      <div styleName='field-customizer'>
+        <Tabs>
+          <TabList>
+            {
+              keys.map((encodingType, i) => {
+                return (
+                  <Tab key={i}>{encodingType}</Tab>
+                );
+              })
+            }
+          </TabList>
+          <div>
+            {
+              keys.map((encodingType, i) => {
+                const customProps = this.getEncodingTypeForChannel()[encodingType] as CustomProp[];
+                return (
+                  <TabPanel key={encodingType + i}>
+                    {
+                      customProps.map(customizableProp => {
+                        const {prop, nestedProp} = customizableProp;
+                        return (
+                          <PropertyEditor
+                            key={JSON.stringify({prop, nestedProp})}
+                            prop={prop}
+                            nestedProp={nestedProp}
+                            shelfId={shelfId}
+                            fieldDef={fieldDef}
+                            handleAction={handleAction}
+                          />
+                        );
+                      })
+                    }
+                  </TabPanel>
+                );
+              })}
+          </div>
+        </Tabs>
       </div>
     );
   }
 
-  private customizableProps() {
-    return [{
-      prop: 'scale',
-      nestedProp: 'type'
-    }];
+  // TODO: Refactor else condition from just mark
+  private getEncodingTypeForChannel() {
```

Comment:
I found the term "encoding type" is very confusing. 

Type = Quantitative / Nominal / ...

Encoding = mapping between encoding channels (x/y/color/...) and fields. 

Overloading another terms for "type" and "encoding" is probably not a good idea. 

Also why "for channel"?  The return result depends on channel, but also depend on other properties (e.g., `type`, for now). 

These are property groups/categories.  So name it `getFieldPropertyGroupIndex`?